File: examples/javascript/basic_completion.mjs
/**
 * Basic Completion Example
 *
 * Demonstrates how to create a client and make a simple chat completion request.
 *
 * Run: node examples/javascript/basic_completion.mjs
 *
 * Requires LITEFORGE_API_KEY environment variable.
 */

import {
  AsyncForgeClient,
  createMessageSystem,
  createMessageUser,
} from '@forge/sdk';

const client = new AsyncForgeClient();
console.log(`Using model: ${client.model}`);
console.log(`Base URL: ${client.baseUrl}`);

const messages = [
  createMessageSystem('You are a helpful assistant. Keep responses brief.'),
  createMessageUser('What is the capital of France?'),
];

const response = await client.complete(messages);
const content = response.choices[0]?.message?.content ?? 'No response';
console.log(`\nResponse: ${content}`);

if (response.usage) {
  console.log(`\nTokens used:`);
  console.log(`  Prompt: ${response.usage.promptTokens}`);
  console.log(`  Completion: ${response.usage.completionTokens}`);
  console.log(`  Total: ${response.usage.totalTokens}`);
}

File: crates/liteforge/examples/basic_completion.rs
//! Basic completion example.
//!
//! Run with: cargo run --example basic_completion

use liteforge::{Message, ForgeClient};

fn main() {
    // Create client (reads LITEFORGE_API_KEY from environment)
    let client = ForgeClient::new();

    println!("Using model: {}", client.model());
    println!("Base URL: {}", client.base_url());

    // Create a simple completion
    let messages = vec![
        Message::system("You are a helpful assistant."),
        Message::user("What is the capital of France?"),
    ];

    match client.complete(messages) {
        Ok(response) => {
            println!("\nResponse:");
            println!("{}", response.content().unwrap_or("No content"));

            if let Some(usage) = &response.usage {
                println!("\nTokens used: {}", usage.total_tokens);
            }
        }
        Err(e) => {
            eprintln!("Error: {}", e);
        }
    }
}
